tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -155,7 +161,13 @@ public static OpenSpan startSpan(String operation, String parentSpanId, SpanType
      */
     @CheckReturnValue
     public static OpenSpan startSpan(String operation, SpanType type) {
-        return getOrCreateCurrentTrace().startSpan(operation, type);
+        Trace trace = getOrCreateCurrentTrace();
+        boolean wasEmpty = trace.isEmpty();
+        OpenSpan result = trace.startSpan(operation, type);
+        if (wasEmpty) {
+            setRequestIdIfPresent(trace.getRequestId());
```

Comment:
This feels like the wrong place to do this. 
I think we should do what carter recommended (https://github.com/palantir/tracing-java/pull/364#issuecomment-558700652) which then means that this only has to happen in one code path.